lesson08/script.js

```diff
@@ -0,0 +1,44 @@
+function createLoading(){
+    const loadingImg = document.createElement('img');
+    loadingImg.src = "loading-circle.gif";
+    loadingImg.alt = "ローディング画像";
+    document.getElementById('js-loading').appendChild(loadingImg);
+}
+
+function removeLoading(){
+    document.getElementById('js-loading').remove();
+}
+
+function createList(value){
+    const fragment = document.createDocumentFragment();
+    for(const list of value){
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+        a.textContent = list.text;
+        a.href = `/${list.to}`;
+        img.src = list.img;
+        img.alt = list.alt;
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    }
+    document.getElementById('js-list').appendChild(fragment);
+}
+
+function getList(){
+    createLoading();
+    return new Promise((resolve,reject) => {
+    const lists = [
+        {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
+        {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+    ];
```

Comment:
IMO
Since this function is a function to get the data of the list,
I thought it was strange that there was data to get in the function. I thought it would be better to declare it outside the function.

Reply:
If make it a global variable, it will be a variable that can be accessed from anywhere.
The longer the code, the harder it is to find the affected area.

Since the "const list" is a variable used only in this function, I thought it was appropriate to declare it in this function.

But, I think function "getlist" name is not appropriate.
I will thinking other function name.

Reply:
Certainly, considering the scope, it's better to declare it here!

It was the function name that made me feel uncomfortable, so I think the function name that means passing list data is appropriate.